File: go-ut-bench/datasets/go/go_code_files_self_contained/simple_function/simple_function_001.go
package main

func ReverseString(s string) string {
	if s == "" {
		return ""
	}
	result := make([]byte, len(s))
	for i := 0; i < len(s); i++ {
		result[i] = s[len(s)-1-i]
	}
	return string(result)
}

func IsPalindrome(s string) bool {
	if s == "" {
		return false
	}
	reversed := ReverseString(s)
	return s == reversed
}

File: go-ut-bench/datasets/cpp/cpp_code_files_self_contained/simple_function/simple_function_001.cpp
#include <stdexcept>
#include <string>
#include <algorithm>

std::string reverseString(const std::string& input) {
    if (input.empty()) {
        return "";
    }
    std::string result = input;
    std::reverse(result.begin(), result.end());
    return result;
}

bool isPalindrome(const std::string& input) {
    if (input.empty()) {
        return false;
    }
    std::string reversed = reverseString(input);
    return input == reversed;
}